### ISSUE TITLE
[Keyboard Navigation>] Keyboard focus is not in logical order under 'Text filter' popup, while navigating using 'tab key.

### ISSUE THREAD

**Issue by GoyalBhuvnesh** [IgniteUI/igniteui-angular#13522]
Keyboard focus is not going in logical order for Product Name header filter. By moving through tab key inside the text filter the focus is not going in logical order.

Shift + Tab Key - going directly to Apply button but require going on text filter button.

source link- [Excel Style Filtering in Angular Data Grid- Ignite UI for Angular (infragistics.com)](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/excel-style-filtering)

Repo Step:

1. Launch the above link.
2. Go to Product Name Header Filter.
3. Navigate to the 'text filter' button.
4. Verify whether keyboard focus is not in logical order under 'Text filter' popup or not.

Actual Result:
Keyboard focus is not in logical order under 'Text filter' popup, while navigating using tab key.
Observation: After pressing 'Shit + tab' key on contain menu keyboard focus moves to 'apply' button and again pressing 'shift + tab' its moves 'cancel 'button.
Expected Result:
Keyboard focus should be in logical order under 'Text filter' popup, while navigating using tab key.
Ex: Keyboard focus should move back to 'text filter' control while navigating using 'shift + tab' key.

**Comment by GoyalBhuvnesh** [comment]
Please provide any solution on this.

**Comment by v-kumarsatya** [comment]
could you please update on this bug?

**Comment by ChronosSF** [comment]
I agree with the expected result. Tab should go through the popup and end on the Search field while Shift+Tab from the popup should get you back on the Text filter field.

We'll look into fixing it during our next sprint, starting Oct 16

**Comment by v-kumarsatya** [comment]
Any update on this bug.

**Comment by ChronosSF** [comment]
A fix is already PRed to the supported branches and we'll try to have it in the patch we are releasing on Monday.

**Comment by ChronosSF** [comment]
Now have in mind that by ARIA spec the secondary dropdown is actually part of the Text Filter component and they work as a single item in terms of the page's tab sequence. So users need to open the Text Filter menu with Enter then navigate it using the arrow keys, not with Tab. Tab and Shift+Tab would only move through the items siblings to Text Filter.

**Comment by v-kumarsatya** [comment]
Request you to share a patch version which you are going to release on coming monday for this bug.

**Comment by v-kumarsatya** [comment]
We have checked on Infragistics portal the latest version it's showing 16.1.X which was updated last on June 2023 so can you please share the version and path link in which you have updated the changes relating to these bugs.
![image](https://github.com/IgniteUI/igniteui-angular/assets/146814222/6c153ac3-4d76-492d-8fcb-a73a3b0c9a5e)

**Comment by ddaribo** [comment]
Hi @v-kumarsatya,

As we have just discussed in a private support case, the fix has already been merged for our 16.1.x branch. Please refer to [this](https://github.com/IgniteUI/igniteui-angular/pull/13591) pull request. And as the dev team has already replied in [this](https://github.com/IgniteUI/igniteui-angular/issues/13522#issuecomment-1782449302) comment, a patch will be released later today. 
Please, follow the [Releases page](https://github.com/IgniteUI/igniteui-angular/releases/), where the new patch version for 16.1.x will be published.

As for the “Update guide” in our documentation, it is rather addressing breaking changes between major versions of the product. Since the fix will be included in a patch and does not qualify as a “breaking change”, information about it will not be included in the Update Guide.

**Comment by ChronosSF** [comment]
@v-kumarsatya , fixes have been released with versions 15.1.32 and 16.1.4 . We'll continue improving the accessibility of the excel style filtering UI this sprint by working on the remaining two items.

**Comment by v-kumarsatya** [comment]
We have verified changes on this Infragistics portal link.
https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/excel-style-filtering
Sample example we have checked.
https://www.infragistics.com/angular-demos/grid/grid-excel-style-filtering-sample-1
But the changes are not showing it behaving like previously so can you please provide link where changes are deployed on Infragistics portal.

**Comment by ChronosSF** [comment]
The samples browser is not updated with each patch. If you open the sample in Stackblitz you'll see the changes as the dependency is a caret range `^16.1.0` and installs the latest patch.

https://stackblitz.com/edit/snfbfx?file=package.json
